core/src/main/java/io/grpc/StatusRuntimeException.java

```diff
@@ -66,4 +68,27 @@ public final Status getStatus() {
   public final Metadata getTrailers() {
     return trailers;
   }
+
+  /**
+   * Wraps the given {@link Throwable} in a {@link StatusRuntimeException}. If it contains an
+   * embedded {@link StatusException} or {@link StatusRuntimeException}, the returned exception will
+   * contain the embedded trailers and status, with the given exception as the cause. Otherwise, an
+   * exception will be generated from an {@link Status#UNKNOWN} status.
+   */
+  @ExperimentalApi
+  public static StatusRuntimeException fromThrowable(Throwable t) {
```

Comment:
I'd rather this not be public. I can't imagine a case our users would need this. It seems stub-specific.